odc/geo/_xr_interop.py

```diff
@@ -123,21 +123,16 @@ def spatial_dims(
 def _mk_crs_coord(crs: CRS, name: str = "spatial_ref") -> xarray.DataArray:
     # pylint: disable=protected-access
 
-    if crs.projected:
-        grid_mapping_name = crs._crs.to_cf().get("grid_mapping_name")
-        if grid_mapping_name is None:
-            grid_mapping_name = "??"
-        grid_mapping_name = grid_mapping_name.lower()
-    else:
-        grid_mapping_name = "latitude_longitude"
-
+    cf = crs.proj.to_cf()
+    cf.setdefault("grid_mapping_name", "??")
     epsg = 0 if crs.epsg is None else crs.epsg
+    crs_wkt = cf.get("crs_wkt", None) or crs.wkt
```

Comment:
```suggestion
    crs_wkt = cf.get("crs_wkt", crs.wkt)
```

Reply:
Reason why I used `or` syntax is because it short-circuits, so we don't need to recompute `wkt` string if it is already included in the cf convention output (it always is, as far as I can tell). It's probably not a very expensive computation, but there is also no telling how often this code might be used in an application.